tests/Backend/WorkspaceCredentialsAssertTrait.php

```diff
@@ -25,24 +25,28 @@ private function assertCredentialsShouldNotWork($connection)
                 $this->getDbConnection($connection);
                 throw new \Exception(self::$RETRY_FAIL_MESSAGE);
             });
-        } catch (\Doctrine\DBAL\Driver\PDOException $e) {
-            // Synapse|Exasol
-            if (!in_array(
-                $e->getCode(),
-                [
-                    //https://docs.microsoft.com/en-us/sql/odbc/reference/appendixes/appendix-a-odbc-error-codes?view=sql-server-ver15
-                    '28000', // Invalid authorization specification
-                    '08004', // Server rejected the connection
-                ],
-                true
-            )) {
-                $this->fail(sprintf('Unexpected error code "%s" for Synapse credentials fail.', $e->getCode()));
+        } catch (\Doctrine\DBAL\Driver\Exception $e) {
+            if ($connection['backend'] === StorageApiTestCase::BACKEND_SYNAPSE) {
+                $this->assertStringContainsString('The login failed', $e->getMessage());
+            }
+            if ($connection['backend'] === StorageApiTestCase::BACKEND_EXASOL) {
+                // Exasol
+                if (!in_array(
+                    (string) $e->getCode(),
+                    [
+                        '28000', // Invalid authorization specification
+                        '08004', // Server rejected the connection
+                    ],
+                    true
+                )) {
+                    $this->fail(sprintf('Unexpected error code "%s" for Exasol credentials fail.', $e->getCode()));
+                }
             }
```

Comment:
neni to tady spis obracene, ze ty kody `28000` a `08004` jsou chybovy kody pro Synapse? (navic je tam odkaz do Microsofti dokumentace)

Reply:
Dobrej point! @zajca ?

Reply:
Aha, nic. Koukám, že v masteru už je nějaký fix mezitím a ten je jiný!

Reply:
jop eště to popravím ten exasol sem testoval, narazil na jiný problém takže sem to nedokončil. Opravím znova ten Synapse case takto to nefunguje.

Reply:
Aha, teď jsem pustil tag... tak ho jdu cancelnout.

Reply:
Bylo by fajn to už nějak dotlačit. Rebase bude pain... když si někdo něco mergne ...

Reply:
jj daj ně eště minutku, musel sem si založit nový TD projekt.

Reply:
přihozeno